File: apps/portal-web/src/utils/environment.js
// Environment-aware URL utilities
export const getEnvironment = () => {
  const hostname = window.location.hostname;
  
  if (hostname.startsWith('uat.') || hostname.includes('-uat.')) {
    return 'uat';
  } else if (hostname.startsWith('live.') || hostname === 'spoton.co.nz' || hostname === 'www.spoton.co.nz' || 
             hostname === 'portal.spoton.co.nz' || hostname === 'staff.spoton.co.nz' || hostname === 'api.spoton.co.nz') {
    return 'live';
  }
  
  // Default to UAT for development
  return 'uat';
};

export const getBaseUrls = () => {
  const env = getEnvironment();
  
  if (env === 'live') {
    return {
      web: 'https://spoton.co.nz',
      portal: 'https://portal.spoton.co.nz',
      staff: 'https://staff.spoton.co.nz',
      api: 'https://api.spoton.co.nz',
      auth: 'https://auth.spoton.co.nz'
    };
  } else {
    return {
      web: 'https://uat.spoton.co.nz',
      portal: 'https://uat.portal.spoton.co.nz',
      staff: 'https://uat.staff.spoton.co.nz',
      api: 'https://uat.api.spoton.co.nz',
      auth: 'https://auth.spoton.co.nz'
    };
  }
};

// Fallback to environment variables if needed
export const getEnvUrl = (service) => {
  const baseUrls = getBaseUrls();
  const envVarMap = {
    web: import.meta.env.VITE_WEB_URL,
    portal: import.meta.env.VITE_PORTAL_URL,
    staff: import.meta.env.VITE_STAFF_URL,
    api: import.meta.env.VITE_API_URL,
    auth: import.meta.env.VITE_KEYCLOAK_URL
  };
  
  return envVarMap[service] || baseUrls[service];
};

// Alias for backward compatibility
export const getBaseUrl = () => {
  const baseUrls = getBaseUrls();
  return baseUrls.api; // Return API URL as the base URL
};

export default {
  getEnvironment,
  getBaseUrls,
  getBaseUrl,
  getEnvUrl
}; 

File: apps/api/core/utils/environment.py
"""
Environment-aware URL utilities for Django backend
"""
import os
from django.conf import settings


def get_environment():
    """
    Determine the current environment based on settings or environment variables
    """
    # Check environment variable first
    env = os.getenv('ENVIRONMENT', '').lower()
    if env in ['live', 'production']:
        return 'live'
    elif env in ['uat', 'staging']:
        return 'uat'
    
    # Fallback to Django DEBUG setting
    if hasattr(settings, 'DEBUG') and not settings.DEBUG:
        return 'live'
    else:
        return 'uat'


def get_base_urls():
    """
    Get environment-aware base URLs
    """
    env = get_environment()
    
    if env == 'live':
        return {
            'web': 'https://spoton.co.nz',
            'portal': 'https://portal.spoton.co.nz',
            'staff': 'https://staff.spoton.co.nz',
            'api': 'https://api.spoton.co.nz',
            'auth': 'https://auth.spoton.co.nz'
        }
    else:
        return {
            'web': 'https://uat.spoton.co.nz',
            'portal': 'https://uat.portal.spoton.co.nz',
            'staff': 'https://uat.staff.spoton.co.nz',
            'api': 'https://uat.api.spoton.co.nz',
            'auth': 'https://auth.spoton.co.nz'
        }


def get_env_url(service):
    """
    Get URL for a specific service based on environment
    """
    base_urls = get_base_urls()
    
    # Check for environment variable override
    env_var_map = {
        'web': 'WEB_URL',
        'portal': 'PORTAL_URL',
        'staff': 'STAFF_URL',
        'api': 'API_URL',
        'auth': 'KEYCLOAK_URL'
    }
    
    env_var = env_var_map.get(service)
    if env_var:
        override_url = os.getenv(env_var)
        if override_url:
            return override_url
    
    return base_urls.get(service, '')


def get_redirect_uris(service):
    """
    Get redirect URIs for Keycloak client configuration
    """
    base_url = get_env_url(service)
    return [f"{base_url}/*"]


def get_web_origins(service):
    """
    Get web origins for Keycloak client configuration
    """
    base_url = get_env_url(service)
    return [base_url]